nautobot_golden_config/templates/nautobot_golden_config/configplan_confirmation.html

```diff
@@ -0,0 +1,288 @@
+                        <div id="diffrender-{{ device_id }}" class="diffrender"></div>
+                    </div>
+                </div>
+            </div>
+        </div>
+        {% endwith %}
+    {% endfor %}
+</div>
+<!-- Deploy Button -->
+{% if perms.extras.run_job %}
+{% include "nautobot_golden_config/job_result_modal.html" with modal_title="Deploy Config Plans" %}
+<button id="startJob" type="button" class="btn btn-primary btn-sm" data-toggle="modal" data-target="#modalPopup" data-config-plan-ids="{{ config_plan_ids }}">
+<span class="mdi mdi-upload-multiple" aria-hidden="true"></span> Deploy Plans
+</button>
+{% endif %}
+{% endblock %}
+{% block javascript %}
+{{ block.super }}
+<script src="{% static 'run_job.js' %}"></script>
+<script src="{% static 'nautobot_golden_config/diff2html-3.4.43/diff2html.min.js' %}"></script>
+<script>
+// Logic for device cards and diff display
+document.addEventListener("DOMContentLoaded", function() {
+    // Function to count the number of added and removed lines in a diff
+    // Currently only counts lines starting with + or -
+    function countDiffs(diffString) {
+        var diffLines = diffString.split("\n");
+        var addedLines = diffLines.filter(function(line) {
+            return line.startsWith("+");
+        });
+        var removedLines = diffLines.filter(function(line) {
+            return line.startsWith("-");
+        });
+        
+        return {
+            added: addedLines.length,
+            removed: removedLines.length
+        };
+    }
+
+    // Select all elements with the collapsable-heading class
+    var collapsableHeadings = document.querySelectorAll(".collapsable-heading");
+
+    // Loop through each collapsable heading and add an event listener that rotates the icon
+    collapsableHeadings.forEach((heading) => {
+        heading.addEventListener("click", function() {
+            var device_id = heading.id.replace("toggle-device-", "");
+            var collapseIcon = document.getElementById("collapse-icon-device-" + device_id);
+            if (collapseIcon) {
+                collapseIcon.classList.toggle("rotated");
+            }
+        });
+    });
+    
+    // Select all elements that start with diffoutput- in the ID
+    var diffContainers = document.querySelectorAll("[id^='diffoutput-']");
+
+    // Loop through each container ultimately rendering each diff
+    diffContainers.forEach(
+        function(container) {
+
+        // Get the data attributes from the container
+        var complianceConfig = container.getAttribute("data-compliance-config");
+        var lastBackup = container.getAttribute("last_backup");
+        var lastIntended = container.getAttribute("last_intended");
```

Comment:
IF either of these doesn't exist in the database the expandable panel that loads on the confirmation page won't dropdown at all.
There is times "config plan type (manual)" where you may not have these.

Reply:
@jeffkala Right - that's why it isn't clickable, there is no changes and no compliance in these cases because you are making a manual change. Could reflect the manual change there I guess but it wouldn't really be a diff..

Reply:
Manual change if you have a backup though I'd expect to see what is changing. Or clickable with "no diff can be presented because xyz", otherwise it gives the feeling that a bug was hit and the app isn't working as expected.

Reply:
I can do clickable with no diff because xyz. That's actually what I had and added in the no click...